src/lib.rs

```diff
@@ -168,12 +168,14 @@ impl Client {
         I: IntoIterator<Item = K>,
         K: AsRef<[u8]>,
     {
-        self.conn.write_all(b"get ").await?;
+        let mut bf = Vec::new();
+        bf.extend(b"get");
         for key in keys {
-            self.conn.write_all(key.as_ref()).await?;
-            self.conn.write_all(b" ").await?;
+            bf.extend(b" ");
+            bf.extend(key.as_ref());
         }
-        self.conn.write_all(b"\r\n").await?;
+        bf.extend(b"\r\n");
```

Comment:
As per my comment on your other pr: https://github.com/Shopify/async-memcached/pull/46#discussion_r1771378315

Writing multiple times to the buffer might be better than creating new data on the stack. (We prefer fewer allocations where possible).

cc @mrattle